### DISCUSSION TITLE
Computing Gradient on Open Surface Mesh

### DISCUSSION THREAD

**Discussion by avaruuser** [libigl/libigl#1939] [Q&A]
Hi all,

I want to compute the gradient of data defined on an open surface mesh. However, during computation of the gradient with igl::grad(V, F, G) I get the error message:

```
Assertion failed: ((i>=0) && ( ((BlockRows==1) && (BlockCols==XprType::ColsAtCompileTime) && i<xpr.rows()) ||((BlockRows==XprType::RowsAtCompileTime) && (BlockCols==1) && i<xpr.cols()))), function Block, file ../libigl/external/eigen/Eigen/src/Core/Block.h, line 122.
[1]    96811 abort      ./script
```

I had the suspicion that this error might occur because the gradient cannot be computed on an open mesh, so I computed the gradient on the corresponding closed surface mesh but I got the exact same error message.
I also tested the gradient computation on the bunny mesh and that worked. Therefore, I guess it must be my specific mesh. Are there any constraints on what the mesh needs to look like? I used a triangulated surface mesh.

I would appreciate any help with this.

**Comment by alecjacobson** [comment]
grad should work fine on an open mesh. Could you share the troublesome
input?

**Reply by avaruuser** [reply]
Interesting, I have shared the input mesh via your e-mail address. Hope that is alright. Thanks
